addons/account/views/account_payment_view.xml

```diff
@@ -242,12 +242,11 @@
                                 <field name="is_internal_transfer" attrs="{'readonly': [('state', '!=', 'draft')]}"/>
                                 <field name="payment_type" widget="radio" options="{'horizontal': True}"
                                         attrs="{'readonly': [('state', '!=', 'draft')]}"/>
-                                <field name="partner_id" context="{'default_is_company': True}" string="Customer"
-                                       attrs="{'readonly':[('state', '!=', 'draft')],
-                                             'invisible':['|', ('partner_type','!=','customer'), ('is_internal_transfer', '=', True)]}"/>
-                                <field name="partner_id" context="{'default_is_company': True}" string="Vendor"
-                                       attrs="{'readonly':[('state', '!=', 'draft')],
-                                               'invisible':['|', ('partner_type','!=','supplier'), ('is_internal_transfer', '=', True)]}"/>
+                                <label for="partner_id" string="Customer" attrs="{'invisible':['|', ('partner_type','!=','customer'), ('is_internal_transfer', '=', True)]}"/>
+                                <label for="partner_id" string="Vendor" attrs="{'invisible':['|', ('partner_type','!=','supplier'), ('is_internal_transfer', '=', True)]}"/>
+                                <field name="partner_id" context="{'default_is_company': True}"
+                                       options="{'no_quick_create': True}"
+                                       attrs="{'readonly':[('state', '!=', 'draft')],'invisible':[('is_internal_transfer', '=', True)]}"/>
```

Comment:
Buggy: two labels are displayed for the field: both the default one and the "Customer" or "Vendor" you're defining here.